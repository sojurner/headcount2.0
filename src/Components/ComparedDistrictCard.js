import React from "react";
import "../CSS/ComparedDistrictCard.css";
import PropTypes from "prop-types";

export const ComparedDistrictCard = ({ location, stats, index }) => {
  return (
    <div className={"compared-district-card" + index}>
      <h1 className="compared-location">{location}</h1>
      {Object.keys(stats).map((year, i) => {
        const avg = stats[year];
        return (
          <section
            key={`avgStats ${i}`}
            className={avg < 0.5 ? "year-red" : "year-black"}
            key={i}
          >
            {year}
            <span className={avg < 0.5 ? "year-red" : "year-black"}>
              ....
              {avg}
            </span>
          </section>
        );
      })}
      <p />
    </div>
  );
};

const { string, objectOf, func, number } = PropTypes;

ComparedDistrictCard.propTypes = {
  location: string,
  stats: objectOf(number),
  selectCard: func.isRequired
};
