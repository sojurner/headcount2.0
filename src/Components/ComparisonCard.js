import React from "react";
import "../CSS/ComparisonCard.css";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div className="compare-data-card">
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] < comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyOne]}
        </p>
      </div>

      <p className="comparison-value">{comparisonData[comparisonKeys[2]]}</p>
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] > comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyTwo]}
        </p>
      </div>
    </div>
  );
};
