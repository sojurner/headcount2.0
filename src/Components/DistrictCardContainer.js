import React from "react";
import PropTypes from "prop-types";
import { DistrictCard } from "./DistrictCard";
import "../CSS/DistrictCardContainer.css";

export const DistrictCardContainer = ({ districts, selectCard }) => {
  let counter = 0;

  const districtCard = districts.map((district, i) => {
    district.clicked ? counter++ : false;
    return (
      <DistrictCard
        key={i}
        {...district}
        selectCard={selectCard}
        counter={counter}
      />
    );
  });
  return <div className="district-card-container"> {districtCard} </div>;
};

const { shape, string, objectOf, number, func, arrayOf, bool } = PropTypes;

DistrictCardContainer.propTypes = {
  districts: arrayOf(
    shape({
      clicked: bool.isRequired,
      location: string,
      stats: objectOf(number)
    })
  ),
  selectCard: func.isRequired
};
