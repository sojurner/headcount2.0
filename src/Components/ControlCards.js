import React from "react";
import { ComparedDistrictCard } from "./ComparedDistrictCard";
import { ComparisonCard } from "./ComparisonCard";
import PropTypes from "prop-types";
import "../CSS/ControlCards.css";

export const ControlCards = ({
  comparisonData,
  selectCard,
  selectedDistricts
}) => {
  const controlCards = Object.keys(comparisonData).map((district, index) => {
    if (selectedDistricts.length === 2) {
      if (comparisonData[district].hasOwnProperty("location")) {
        return (
          <ComparedDistrictCard
            key={`compared ${index}`}
            selectCard={selectCard}
            location={district}
            stats={comparisonData[district].stats}
            index={index}
          />
        );
      }
      return (
        <ComparisonCard key={index} comparisonData={comparisonData[district]} />
      );
    }
  });
  return <div className="control-cards">{controlCards} </div>;
};

const { string, number, bool, shape, objectOf, func } = PropTypes;

ControlCards.propTypes = {
  comparisonData: objectOf(
    shape({
      location: string,
      stats: objectOf(number),
      clicked: bool
    })
    // comparisonData: objectOf(shape({
    //   compared: number
    // }))
  ),
  selectCard: func.isRequired
};
