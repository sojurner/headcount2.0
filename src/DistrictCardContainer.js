import React from "react";
import DistrictRepository from "./helper";
import kindergarners from "./data/kindergartners_in_full_day_program";
import { DistrictCard } from "./DistrictCard";
import "./CSS/DistrictCardContainer.css";

this.districtRepository = new DistrictRepository(kindergarners);

export const DistrictCardContainer = ({ districts }) => {
  console.log(districts);
  const districtCard = districts.map((district, i) => {
    return <DistrictCard key={i} {...district} />;
  });
  return <div className="district-card-container"> {districtCard} </div>;
};
