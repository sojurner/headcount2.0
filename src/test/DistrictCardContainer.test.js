import React from "react";
import { DistrictCardContainer } from "../Components/DistrictCardContainer";

describe("DistrictCardContainer", () => {
  let shallowWrap;
  let mockDistrictsOne = [
    {
      location: "COLORADO",
      clicked: true,
      stats: {
        2004: 0.24,
        2005: 0.278
      }
    },
    {
      location: "ACADEMY 20",
      clicked: true,
      stats: {
        2004: 0.302,
        2005: 0.267
      }
    }
  ];

  let mockDistrictsTwo = [
    {
      location: "COLORADO",
      clicked: false,
      stats: {
        2004: 0.24,
        2005: 0.278
      }
    },
    {
      location: "ACADEMY 20",
      clicked: false,
      stats: {
        2004: 0.302,
        2005: 0.267
      }
    }
  ];

  beforeEach(() => {
    shallowWrap = shallow(
      <DistrictCardContainer
        districts={mockDistrictsOne}
        selectCard={jest.fn()}
      />
    );
  });

  it("should exist", () => {
    expect(shallowWrap).toBeDefined();
  });

  it("should increment counter for each Card that has clicked property equal to true", () => {
    let counter = shallowWrap.props().children.map(child => child[1])[1].props
      .counter;
    expect(counter).toEqual(2);
  });

  it("should match snap shot with an empty District Card container", () => {
    shallowWrap = shallow(
      <DistrictCardContainer districts={[]} selectCard={jest.fn()} />
    );
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot with an array of districts with not clicked property ", () => {
    shallowWrap = shallow(
      <DistrictCardContainer
        districts={mockDistrictsTwo}
        selectCard={jest.fn()}
      />
    );
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot with districts with clicked property", () => {
    expect(shallowWrap).toMatchSnapshot();
  });
});
