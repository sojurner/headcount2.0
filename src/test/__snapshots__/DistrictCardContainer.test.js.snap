// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCardContainer should match snap shot with an empty District Card container 1`] = `
<div
  className="district-card-container"
>
   
   
</div>
`;

exports[`DistrictCardContainer should match snapshot with an array of districts with not clicked property  1`] = `
<div
  className="district-card-container"
>
   
  <Component
    clicked={false}
    counter={0}
    key="0"
    location="COLORADO"
    selectCard={[Function]}
    stats={
      Object {
        "2004": 0.24,
        "2005": 0.278,
      }
    }
  />
  <Component
    clicked={false}
    counter={0}
    key="1"
    location="ACADEMY 20"
    selectCard={[Function]}
    stats={
      Object {
        "2004": 0.302,
        "2005": 0.267,
      }
    }
  />
   
</div>
`;

exports[`DistrictCardContainer should match snapshot with districts with clicked property 1`] = `
<div
  className="district-card-container"
>
   
  <Component
    clicked={true}
    counter={1}
    key="0"
    location="COLORADO"
    selectCard={[Function]}
    stats={
      Object {
        "2004": 0.24,
        "2005": 0.278,
      }
    }
  />
  <Component
    clicked={true}
    counter={2}
    key="1"
    location="ACADEMY 20"
    selectCard={[Function]}
    stats={
      Object {
        "2004": 0.302,
        "2005": 0.267,
      }
    }
  />
   
</div>
`;
